plugins/inputs/suricata/suricata.go

```diff
@@ -150,27 +154,35 @@ func flexFlatten(outmap map[string]interface{}, field string, v interface{}, del
 		}
 	case float64:
 		outmap[field] = v.(float64)
+	case string:
+		outmap[field] = v.(string)
 	default:
 		return fmt.Errorf("unsupported type %T encountered", t)
 	}
 	return nil
 }
 
-func (s *Suricata) parse(acc telegraf.Accumulator, sjson []byte) {
-	// initial parsing
-	var result map[string]interface{}
-	err := json.Unmarshal(sjson, &result)
-	if err != nil {
-		acc.AddError(err)
+func (s *Suricata) parseAlert(acc telegraf.Accumulator, result map[string]interface{}) {
+	if _, ok := result["alert"].(map[string]interface{}); !ok {
+		s.Log.Debug("The 'alert' sub-object does not have required structure")
 		return
 	}
 
-	// check for presence of relevant stats
-	if _, ok := result["stats"]; !ok {
-		s.Log.Debug("Input does not contain necessary 'stats' sub-object")
-		return
+	totalmap := make(map[string]interface{})
+	for k, v := range result["alert"].(map[string]interface{}) {
+		//source and target fields are maps
+		err := flexFlatten(totalmap, k, v, s.Delimiter)
+		if err != nil {
+			s.Log.Debug(err.Error())
```

Comment:
Can you maybe state in the message where the problem occurred, like
```suggestion
			s.Log.Debugf("Flattening alert failed: %v", err)
```

Reply:
Agree